pallets/liquid-staking/src/lib.rs

```diff
@@ -785,194 +751,70 @@ pub mod pallet {
-                                            value,
-                                            payee: payee.clone(),
-                                        },
-                                    )),
-                                },
-                            ))),
-                        ],
-                    })));
-
-                let msg = Self::ump_transact(call.encode().into(), Self::xcm_weight().bond_weight)?;
-
-                match T::XcmSender::send_xcm(MultiLocation::parent(), msg) {
-                    Ok(()) => {
-                        Self::deposit_event(Event::<T>::Bonding(controller, value, payee));
-                    }
-                    Err(_e) => {
-                        return Err(Error::<T>::BondFailed.into());
-                    }
-                }
-            });
-
-            Ok(())
+            T::XCM::bond_internal(
+                value,
+                payee,
+                Self::derivative_para_account_id(),
+                Self::xcm_weight().bond_weight,
+                T::AccountIdToMultiLocation::convert(Self::para_account_id()),
+                Self::staking_currency()?,
+                Self::account_id(),
+                T::DerivativeIndex::get(),
+            )
         }
 
         #[require_transactional]
         fn bond_extra_internal(value: BalanceOf<T>) -> DispatchResult {
-            let stash = T::Lookup::unlookup(Self::derivative_para_account_id());
-
-            switch_relay!({
-                let call =
-                    RelaychainCall::Utility(Box::new(UtilityCall::BatchAll(UtilityBatchAllCall {
-                        calls: vec![
-                            RelaychainCall::Balances(BalancesCall::TransferKeepAlive(
-                                BalancesTransferKeepAliveCall { dest: stash, value },
-                            )),
-                            RelaychainCall::Utility(Box::new(UtilityCall::AsDerivative(
-                                UtilityAsDerivativeCall {
-                                    index: T::DerivativeIndex::get(),
-                                    call: RelaychainCall::Staking::<T>(StakingCall::BondExtra(
-                                        StakingBondExtraCall { value },
-                                    )),
-                                },
-                            ))),
-                        ],
-                    })));
-
-                let msg =
-                    Self::ump_transact(call.encode().into(), Self::xcm_weight().bond_extra_weight)?;
-
-                match T::XcmSender::send_xcm(MultiLocation::parent(), msg) {
-                    Ok(()) => {
-                        Self::deposit_event(Event::<T>::BondingExtra(value));
-                    }
-                    Err(_e) => {
-                        return Err(Error::<T>::BondExtraFailed.into());
-                    }
-                }
-            });
-
-            Ok(())
+            T::XCM::bond_extra_internal(
```

Comment:
same as above

Reply:
@GopherJ Can we do this as a part of separate issue?